cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -313,6 +322,7 @@ void create(String namespaceId, String name, DatasetInstanceConfiguration props)
    */
   void update(DatasetId instance, Map<String, String> properties) throws Exception {
     ensureNamespaceExists(instance.getParent());
+    ensureAccess(instance);
```

Comment:
That will expose the existence of the dataset instance, which is not we want, right?